src/apps/bridge/reportBuilder.cpp

```diff
@@ -0,0 +1,490 @@
+        printf("Failed to add sample member in addSamplesToReport\n");
+      }
+      if (sensor_report_encoder_add_sample_member(context, encode_double_sample_member, &aanderaa_sample.direction_circ_mean_rad) != CborNoError) {
+        printf("Failed to add sample member in addSamplesToReport\n");
+      }
+      if (sensor_report_encoder_add_sample_member(context, encode_double_sample_member, &aanderaa_sample.direction_circ_std_rad) != CborNoError) {
+        printf("Failed to add sample member in addSamplesToReport\n");
+      }
+      if (sensor_report_encoder_add_sample_member(context, encode_double_sample_member, &aanderaa_sample.temp_mean_deg_c) != CborNoError) {
+        printf("Failed to add sample member in addSamplesToReport\n");
+      }
+      if (sensor_report_encoder_close_sample(context) != CborNoError) {
+        printf("Failed to close sample in addSamplesToReport\n");
+      }
+      break;
+    }
+    default: {
+      printf("Received invalid sensor type in addSamplesToReport\n");
+      break;
+    }
+  }
+
+}
+
+// Task init
+void reportBuilderInit(cfg::Configuration* sys_cfg) {
+  configASSERT(sys_cfg);
+  _ctx._samplesPerReport = DEFAULT_SAMPLES_PER_REPORT;
+  sys_cfg->getConfig(AppConfig::SAMPLES_PER_REPORT, strlen(AppConfig::SAMPLES_PER_REPORT), _ctx._samplesPerReport);
+  _ctx._transmitAggregations = DEFAULT_TRANSMIT_AGGREGATIONS;
+  sys_cfg->getConfig(AppConfig::TRANSMIT_AGGREGATIONS, strlen(AppConfig::TRANSMIT_AGGREGATIONS), _ctx._transmitAggregations);
+  _ctx._sample_counter = 0;
+  _report_builder_queue =
+      xQueueCreate(REPORT_BUILDER_QUEUE_SIZE, sizeof(report_builder_queue_item_t));
+  configASSERT(_report_builder_queue);
+  // create task
+  BaseType_t rval = xTaskCreate(report_builder_task, "REPORT_BUILDER", 1024, NULL,
+                                REPORT_BUILDER_TASK_PRIORITY, NULL);
+  configASSERT(rval == pdPASS);
+}
+
+// Task
+static void report_builder_task(void *parameters) {
+  (void) parameters;
+
+  report_builder_queue_item_t item;
+
+  while (1) {
+    if (xQueueReceive(_report_builder_queue, &item,
+                      portMAX_DELAY) == pdPASS) {
+      switch (item.message_type) {
+        case REPORT_BUILDER_INCREMENT_SAMPLE_COUNT: {
+          _ctx._sample_counter++;
+          printf("Incrementing sample counter to %d\n", _ctx._sample_counter);
+          if (_ctx._sample_counter >= _ctx._samplesPerReport) {
+            if (_ctx._samplesPerReport > 0 && _ctx._transmitAggregations) {
+              // Need to have more than one node to report anything (1 node would be just the bridge)
+              if (_ctx._report_period_num_nodes > 1) {
+                uint32_t num_sensors = _ctx._report_period_num_nodes - 1; // Minus one since topo includes the bridge
+                sensor_report_encoder_context_t context;
+                uint8_t cbor_buffer[315];
```

Comment:
Please #define or constexpr `315`

Reply:
This is also a fairly large buffer to pop onto the task stack - it looks like the stack depth is 1024 so this buffer consumes almost 1/3 of the stack here. 

Watch the watermark carefully or consider dynamic allocation or `static & memset(0)` so we don't get a stack overflow scenario.

Reply:
Good point! I'll go with dynamic allocation since we may want to support multipart messages in the future.